Linker error in `make check` on fresh machine

I'm getting a linker error during a `make check` that I can only reproduce on a fresh linux machine.  'Fresh linux machine' meaning I built `gcc`, `gsl`, `mpich` and `boost` all from source on a brand new linux machine.

Here's the error:

```
/bin/bash ../libtool  --tag=CXX   --mode=link mpic++  -g -O2 -Wall   -o test_GaussianMean1DRegression test_GaussianMean1DRegression.o  -L../src/ -lqueso 
libtool: link: mpic++ -g -O2 -Wall -o .libs/test_GaussianMean1DRegression test_GaussianMean1DRegression.o  -L../src/ /home/damon/git/queso/src/.libs/libqueso.so -Wl,-rpath -Wl,/home/damon/queso/lib
/usr/bin/ld: test_GaussianMean1DRegression.o: undefined reference to symbol '_ZN5boost15program_options13variables_mapC1Ev'
/home/damon/sw/lib/libboost_program_options.so.1.55.0: error adding symbols: DSO missing from command line
```

I don't see this on any of the SL6 machines at ICES.  A possible fix is the following:

``` diff
diff --git a/test/Makefile.am b/test/Makefile.am
index 3f4e48a..6526b4f 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -30,7 +30,7 @@ check_PROGRAMS += test_SequenceOfVectorsErase
 check_PROGRAMS += test_GaussianMean1DRegression
 check_PROGRAMS += test_gpmsa_cobra

-LIBS         = -L$(top_builddir)/src/ -lqueso
+LIBS         = -L$(top_builddir)/src/ -lqueso $(BOOST_PROGRAM_OPTIONS_LDFLAGS) $(BOOST_PROGRAM_OPTIONS_LIBS)

 AM_CPPFLAGS =
 AM_CPPFLAGS += -I.
```

I don't like this solution.  I feel like I should be able to compile a test using boost program options just by linking to queso.  I shouldn't need to explicitly link to boost program options.

--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -81,6 +81,7 @@ test_inf_gaussian_SOURCES = $(top_srcdir)/test/test_infinite/test_inf_gaussian.C
 test_inf_options_SOURCES = $(top_srcdir)/test/test_infinite/test_inf_options.C
 test_SequenceOfVectorsErase_SOURCES = $(top_srcdir)/test/test_SequenceOfVectors/test_SequenceOfVectorsErase.C
 test_GaussianMean1DRegression_SOURCES = $(top_srcdir)/test/test_Regression/test_GaussianMean1DRegression.C
+test_GaussianMean1DRegression_LDFLAGS = $(BOOST_PROGRAM_OPTIONS_LDFLAGS) $(BOOST_PROGRAM_OPTIONS_LIBS)
 test_gpmsa_cobra_SOURCES = $(top_srcdir)/test/test_Regression/test_gpmsa_cobra.C
 
 # Files to freedom stamp

